sqlglot/parser.py

```diff
@@ -2251,7 +2247,19 @@ def _parse_property(self) -> t.Optional[exp.Expression]:
             return self._parse_sortkey(compound=True)
 
         if self._match_text_seq("SQL", "SECURITY"):
-            return self.expression(exp.SqlSecurityProperty, definer=self._match_text_seq("DEFINER"))
+            if self._match_text_seq("DEFINER"):
+                return self.expression(
+                    exp.SqlSecurityProperty,
+                    definer=True,
+                )
+
+            if self._match_text_seq("INVOKER"):
+                return self.expression(
+                    exp.SqlSecurityProperty,
+                    invoker=True,
+                )
+
+            return self.expression(exp.SqlSecurityProperty)
```

Comment:
Similarly, we can simplify all this section as:


```suggestion
            this = self._match_texts(("DEFINER", "INVOKER")) and self._prev.text.upper()

            return self.expression(exp.SqlSecurityProperty, this=this)
```